src/badge/withBadge.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+import Badge from './Badge';
+
+const styles = StyleSheet.create({
+  badge: {
+    borderRadius: 9,
+    height: 18,
+    minWidth: 0,
+    width: 18
+  },
+  badgeContainer: {
+    position: 'absolute',
+    shadowColor: 'black',
+    shadowOffset: {
+      width: 0,
+      height: 0.5
+    },
+    shadowRadius: 0.5,
+    shadowOpacity: 1.0
+  },
+  badgeText: {
+    fontSize: 10,
+    paddingHorizontal: 0
+  }
+});
+
+const withBadge = (value, options = {}) => WrappedComponent => {
+  class WithBadge extends React.Component {
+    render() {
+      const {
+        status = 'error',
```

Comment:
Don't duplicate props from `Badge`, badge already has a `status`, `onPress` and `value` prop. If anything, pull it from badgeProps

Reply:
Maybe an easier design for the props would be to let it accept all the badgeProps:
```js
left,
bottom,
top,
right,
hidden,
...badgeProps
```

Reply:
You mean like this:

```jsx
const {
        left = 0,
        bottom = 0,
        top = -5,
        right = 8,
        hidden = !value,
        badgeProps = {
          status = "error",
          onPress
        },
      } = options;
      const badgeValue = typeof value === "function" ? value(this.props) : value;
      return (
        <View>
          <WrappedComponent {...this.props} />
          {!hidden && (
            <Badge
              badgeStyle={styles.badge}
              textStyle={styles.badgeText}
              value={badgeValue}
              status={status}
              containerStyle={[styles.badgeContainer, { bottom, left, right, top }]}
              onPress={onPress}
              {...badgeProps}
            />
          )}
        </View>
      );
```

Reply:
As in they'll have no use for `badgeProps` option, any properties outside left, bottom, top, right, hidden, would be considered badgeProps

Reply:
The problem is how would you define default values then if you use the rest operator?

```js
const { left = 0, bottom = 0, top = -5, right = 8, hidden = !value, ...badgeProps = {status: "error", onPress} } = options;
```
Isn't possible.

Reply:
Okay I got it working implementing right now...

Reply:
Done.